src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -1163,14 +1165,16 @@ func newAttachedResources(ctx context.Context, r *ClusterReconciler, log logr.Lo
 	}
 }
 
+type resourceKey string
+
 func (a *attachedResources) Ensure() (ctrl.Result, error) {
 	result := ctrl.Result{}
 	var errs error
 	for key, resource := range a.items {
 		if resource == nil {
 			continue
 		}
-		err := resource.Ensure(a.ctx)
+		err := resource.Ensure(context.WithValue(a.ctx, resourceKey("resource"), key))
```

Comment:
I'm not sure about creating context with resource because reconciliation context should have this information

Reply:
I had to do this to solve a race that was found in testing.

Reply:
Interesting!